File: Happy Number/file.py
def happy(n):
    x=n 
    while(x!=1 and x!=4):
        sumval=0
        while(x>0):
            sumval=sumval+(x%10)*(x%10)
            x=x//10
        x=sumval
    return x


num=int(input())
flag=happy(num)
print("Yes it is a Happy Number.") if(flag==1) else print("No it is not a Happy Number")

File: Happy Number/file.c
#include<stdio.h>

int happy(int n){
    
    int x=n;
    
    while(x!=1 && x!=4){
        int sum=0;
        while(x>0){
            sum=sum+(x%10)*(x%10);
            x/=10;
        }
        x=sum;
    }
    return x;
}
int main()
{
    int n;
    scanf("%d",&n);
    int flag=happy(n);
    flag==1 ? printf("Yes it is a Happy Number.") : printf("No it is not a Happy Number");
}
